app.rb

```diff
@@ -44,14 +44,8 @@ def create_student
     print 'Has parent permission? [Y/N]: '
     permission = gets.chomp.upcase
 
-    case permission
-    when 'N'
-      student = Student.new(age, 'classroom', name: name, parent_permission: false)
-      @people << student
-    when 'Y'
-      student = Student.new(age, 'classroom', name: name, parent_permission: true)
-      @people << student
-    end
+    student = Student.new(age, 'classroom', name: name, parent_permission: permission == 'Y')
+    @people << student
     puts 'Student created successfully'
   end
```

Comment:
- [ ] I would like to reiterate what the previous reviewer stated. All codes responsible for receiving inputs or codes responsible for validating the input from the user should be put in the main.rb file or a separate class or class method or module. The other methods for creating a class instance or listing instances should only create those instances. 👍 😎 

---

For Example, your `create_book` method should only create a new instance of the `Book` class while codes in the method for getting the `title` and `author` of the _**book**_  from the console app should be put in a separate method, class or module. In the same vein, in the `create_teacher` method, all codes that are responsible for getting the teacher's `name`, `age` and `specialization` should be put in a separate class, module or method while the `create_teacher` method should be responsible for just creating a new instance of the `Teacher` class. 💻 

--- 

Kindly apply these changes for the following methods of the `App` class in the `app.rb` file:

- `create_person`
- `create_student`
- `create_teacher`
- `create_book`
- `create_rental`
- `list_all_rentals`

---

#### N.B: Please ensure that the app works as it should after refactoring the code. 👍

Reply:
done.